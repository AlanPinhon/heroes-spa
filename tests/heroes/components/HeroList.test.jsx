import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { HeroList } from '../../../src/heroes/components';

//Test realizado como pr√°ctica fuera del curso

describe('Pruebas en <HeroList/>', () => {

  test('debe de mostrar la lista de personajes de Marvel y hacer match con el snapshot', () => {

    const {container} = render(
      <MemoryRouter>
        <HeroList publisher={'Marvel Comics'}/>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();

  })

})