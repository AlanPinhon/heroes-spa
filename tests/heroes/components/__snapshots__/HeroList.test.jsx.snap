// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroList/> debe de mostrar la lista de personajes de Marvel y hacer match con el snapshot 1`] = `
<div>
  <div
    class="row rows-cols-1 row-cols-md-3 g-3"
  >
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Spider Man"
              class="card-img"
              src="/assets/heroes/marvel-spider.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Spider Man
              </h5>
              <p
                class="card-text"
              >
                Peter Parker
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Amazing Fantasy #15
                </small>
              </p>
              <a
                href="/hero/marvel-spider"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Captain America"
              class="card-img"
              src="/assets/heroes/marvel-captain.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Captain America
              </h5>
              <p
                class="card-text"
              >
                Steve Rogers
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Captain America Comics #1
                </small>
              </p>
              <a
                href="/hero/marvel-captain"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Iron Man"
              class="card-img"
              src="/assets/heroes/marvel-iron.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Iron Man
              </h5>
              <p
                class="card-text"
              >
                Tony Stark
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Tales of Suspense #39
                </small>
              </p>
              <a
                href="/hero/marvel-iron"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Thor"
              class="card-img"
              src="/assets/heroes/marvel-thor.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Thor
              </h5>
              <p
                class="card-text"
              >
                Thor Odinson
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Journey into Myster #83
                </small>
              </p>
              <a
                href="/hero/marvel-thor"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Hulk"
              class="card-img"
              src="/assets/heroes/marvel-hulk.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Hulk
              </h5>
              <p
                class="card-text"
              >
                Bruce Banner
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  The Incredible Hulk #1
                </small>
              </p>
              <a
                href="/hero/marvel-hulk"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Wolverine"
              class="card-img"
              src="/assets/heroes/marvel-wolverine.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Wolverine
              </h5>
              <p
                class="card-text"
              >
                James Howlett
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  The Incredible Hulk #180
                </small>
              </p>
              <a
                href="/hero/marvel-wolverine"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Daredevil"
              class="card-img"
              src="/assets/heroes/marvel-daredevil.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Daredevil
              </h5>
              <p
                class="card-text"
              >
                Matthew Michael Murdock
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Daredevil #1
                </small>
              </p>
              <a
                href="/hero/marvel-daredevil"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Hawkeye"
              class="card-img"
              src="/assets/heroes/marvel-hawkeye.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Hawkeye
              </h5>
              <p
                class="card-text"
              >
                Clinton Francis Barton
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  Tales of Suspense #57
                </small>
              </p>
              <a
                href="/hero/marvel-hawkeye"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Cyclops"
              class="card-img"
              src="/assets/heroes/marvel-cyclops.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Cyclops
              </h5>
              <p
                class="card-text"
              >
                Scott Summers
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  X-Men #1
                </small>
              </p>
              <a
                href="/hero/marvel-cyclops"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col animate__animated animate__fadeIn"
    >
      <div
        class="card"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-4"
          >
            <img
              alt="Silver Surfer"
              class="card-img"
              src="/assets/heroes/marvel-silver.jpg"
            />
          </div>
          <div
            class="col-8"
          >
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              >
                Silver Surfer
              </h5>
              <p
                class="card-text"
              >
                Norrin Radd
              </p>
              <p
                class="card-text"
              >
                <small
                  class="text-muted"
                >
                  The Fantastic Four #48
                </small>
              </p>
              <a
                href="/hero/marvel-silver"
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
